mmcls/models/necks/gem.py

```diff
@@ -28,14 +28,20 @@ class GeneralizedMeanPooling(nn.Module):
             Default: 1e-6
         clamp (bool): Use clamp before pooling.
             Default: True
+        p_trainable (bool): Toggle whether Parameter p is trainable or not.
+            Default: True
     """
 
-    def __init__(self, p=3., eps=1e-6, clamp=True):
+    def __init__(self, p=3., eps=1e-6, clamp=True, p_trainable=True):
         assert p >= 1, "'p' must be a value greater then 1"
         super(GeneralizedMeanPooling, self).__init__()
-        self.p = Parameter(torch.ones(1) * p)
+        if p_trainable:
+            self.p = Parameter(torch.ones(1) * p)
```

Comment:
This method may influence the checkpoint loading if `p_trainable=False`.
Can we use `requries_grad=p_trainable` here?

Reply:
I made the following changes.

`self.p = Parameter(torch.ones(1) * p, requires_grad=p_trainable)`

I would like to ask, just for reference, where exactly in checkpoint loading might be affected?